ambry-network/src/main/java/com.github.ambry.network/NettyServerRequestResponseChannel.java

```diff
@@ -32,6 +33,13 @@ public NettyServerRequestResponseChannel(int queueSize) {
   /** Send a request to be handled, potentially blocking until there is room in the queue for the request */
   @Override
   public void sendRequest(NetworkRequest request) throws InterruptedException {
+    DataInputStream stream = new DataInputStream(request.getInputStream());
+    try {
+      // TODO: Where is this done in socket channel?
```

Comment:
This is done inside of the selector, reading the size of the payload is implemented in `BoundedReceive`